src/nnue/features/feature_set.h

```diff
@@ -43,90 +43,6 @@ namespace Eval::NNUE::Features {
   template <typename Derived>
   class FeatureSetBase {
 
```

Comment:
Since now FeatureSetBase is empty and FeatureSet only has some constants it looks like they are not needed at all. Though complete removal should be pondered (keeping it allows some static_asserts). For now we should at least leave a note inside the FeatureSetBase and FeatureSet that the current code is optimized for a single architecture and to make it work with arbitrary architectures it's required to bring back the code from https://github.com/official-stockfish/Stockfish/blob/560c776397483feaaa0deb5b666f46ff3f5b655f/src/nnue/features/feature_set.h#L44-L173